L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,293 @@
+  outJSON["TriggerGeometryConfig"]["ScintillatorLinksPerModule"] = scintillator_links_per_module_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorModuleSize"] = scintillator_module_size_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorTriggerCellSize"] = scintillator_trigger_cell_size_;
+
+  // fill truncation parameters
+  outJSON["TruncationConfig"]["rozMin"] = roz_min_;
+  outJSON["TruncationConfig"]["rozMax"] = roz_max_;
+  outJSON["TruncationConfig"]["rozBins"] = roz_bins_;
+  outJSON["TruncationConfig"]["maxTcsPerBin"] = max_tcs_per_bins_;
+  outJSON["TruncationConfig"]["phiSectorEdges"] = phi_edges_;
+
+  // fill trigger geometry
+  fillTriggerGeometry(outJSON);
+
+  // write out JSON file
+  std::ofstream outputfile(outJSONname_.c_str());
+  outputfile << std::setw(4) << outJSON << std::endl;
+}
+
+void HGCalBackendStage1ParameterExtractor::fillTriggerGeometry(json& json_file) {
+  trigger_map_set trigger_cells;
+
+  // retrieve valid trigger cells
+  std::cout << "Getting EE trigger cells\n";
+  for (const auto& id : triggerGeometry_->eeGeometry()->getValidDetIds()) {
+    HGCSiliconDetId detid(id);
+    if (!triggerGeometry_->eeTopology().valid(id))
+      continue;
+    uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+    auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+    itr_insert.first->second.emplace(id);
+  }
+  std::cout << "Getting HSi trigger cells\n";
+  for (const auto& id : triggerGeometry_->hsiGeometry()->getValidDetIds()) {
+    HGCSiliconDetId detid(id);
+    if (!triggerGeometry_->hsiTopology().valid(id))
+      continue;
+    uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+    auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+    itr_insert.first->second.emplace(id);
+  }
+  std::cout << "Getting HSc trigger cells\n";
+  for (const auto& id : triggerGeometry_->hscGeometry()->getValidDetIds()) {
+    uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+    auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+    itr_insert.first->second.emplace(id);
+  }
+  if (triggerGeometry_->isWithNoseGeometry()) {
+    std::cout << "Getting NOSE trigger cells\n";
+    for (const auto& id : triggerGeometry_->noseGeometry()->getValidDetIds()) {
+      uint32_t trigger_cell = triggerGeometry_->getTriggerCellFromCell(id);
+      auto itr_insert = trigger_cells.emplace(trigger_cell, std::unordered_set<uint32_t>());
+      itr_insert.first->second.emplace(id);
+    }
+  }
+
+  // loop over trigger cells
+  edm::LogPrint("JSONFilling") << "Filling JSON map";
+  for (const auto& triggercell_cells : trigger_cells) {
+    DetId id(triggercell_cells.first);
+    // get module ID and check if relevant module (sector0, zside=-1)
+    uint32_t moduleId = triggerGeometry_->getModuleFromTriggerCell(id);
+    if (moduleId == 0)
+      continue;
```

Comment:
It should be checked if the module is disconnected or not, like it is done there:
https://github.com/hgc-tpg/cmssw/blob/9bff0b25ff7a1ebf947c8ef0043ea00e35efb07c/L1Trigger/L1THGCal/plugins/veryfrontend/HGCalVFEProcessorSums.cc#L50
It's actually done later by looking at the number of lpgbt links, but it is cleaner to use the dedicated function in the trigger geometry

Reply:
done, and removed the lpgbt link check